src/shell/autoexec.cpp

```diff
@@ -693,9 +693,10 @@ void AutoExecModule::AutoMountDetectedDrive(const std::string& dir_letter)
 	// See if there are mountable images in the path
 	const auto image_paths = GetMountableImagesPaths(drive_path);
 
-	// Setting the drive type to 'dir' will force dosbox to mount the path
-	// as a directory, even if there are mountable images in the path
-	if (image_paths.empty() || settings->type == "dir") {
+	// Explicitly setting the drive type to anything but 'cdrom' will force
+	// dosbox to mount the path as a directory, even if there are mountable
+	// images in the path
+	if (image_paths.empty() || (!settings->type.empty() && settings->type != "cdrom")) {
```

Comment:
Agree with the commit description, we don't want to complicate things for niche usages like that. Good call!